include/storage/io.hpp

```diff
@@ -60,29 +61,31 @@ class FileReader
 
     std::size_t GetSize()
     {
-        const boost::filesystem::ifstream::pos_type position = input_stream.tellg();
-        input_stream.seekg(0, std::ios::end);
-        const boost::filesystem::ifstream::pos_type file_size = input_stream.tellg();
-
-        if (file_size == boost::filesystem::ifstream::pos_type(-1))
+        const boost::filesystem::path path(filepath);
+        try
         {
-            throw util::RuntimeError("Unable to determine file size for " +
-                                         std::string(filepath.string()),
-                                     ErrorCode::FileIOError,
-                                     SOURCE_REF,
-                                     std::strerror(errno));
+            return std::size_t(boost::filesystem::file_size(path)) -
+                   ((fingerprint == FingerprintFlag::VerifyFingerprint) ? sizeof(util::FingerPrint)
+                                                                        : 0);
         }
-
-        // restore the current position
-        input_stream.seekg(position, std::ios::beg);
-
-        if (fingerprint == FingerprintFlag::VerifyFingerprint)
+        catch (boost::filesystem::filesystem_error &ex)
         {
-            return std::size_t(file_size) - sizeof(util::FingerPrint);
+            std::cout << ex.what() << std::endl;
+            throw;
         }
-        else
+    }
+
+    /* Read one line */
+    template <typename T> void ReadLine(T *dest, const std::size_t count)
+    {
+        if (0 < count)
         {
-            return file_size;
+            const auto &ios =
+                input_stream.getline(reinterpret_cast<char *>(dest), count * sizeof(T));
+            for (std::size_t n = ios.gcount(); n < count; ++n)
+            {
+                reinterpret_cast<char *>(dest)[n] = '\0';
```

Comment:
Can we init string with this value on `resize()`?

Reply:
Thanks for your comment. I got it. I will update my code.

Reply:
The dest is not std::string but template T. So, I could not apply resize().
instead of that, I modified to use memset.
> memset(dest, 0, count * sizeof(T));
Is this satisfy your intention?